### ISSUE TITLE
Add support for expressions that are not statements

### ISSUE THREAD

**Issue by chucklepie** [PyCQA/pylint#3731]
Unless there is a flag defaulting to off that I have not found, pylint does not perform a check on statements that are expressions that do nothing and are possibly written incorrectly by mistake.

For example:

```
a == 6
```

This is clearly meant to be 'a=6', and is hard to find in a code file, but not picked up. There is never a use case where this was intended.

As an example, C# parser solves this by flagging all lines that are not statements, incrementers or expressions creating a new object.

**Comment by AWhetter** [comment]
The `pointless-statement` warning covers this case. Do you have a configuration disabling this warning? It's enabled by default.
